packages/grid_client/tests/modules/applications/discourse.test.ts

```diff
@@ -0,0 +1,254 @@
+  expect(res.contracts.updated).toHaveLength(0);
+  expect(res.contracts.deleted).toHaveLength(0);
+
+  const vmsList = await gridClient.machines.list();
+  log(vmsList);
+
+  //VM List Assertions
+  expect(vmsList.length).toBeGreaterThanOrEqual(1);
+  expect(vmsList).toContain(vms.name);
+
+  const result = await gridClient.machines.getObj(vms.name);
+  log(result);
+
+  //VM Assertions
+  expect(result[0].nodeId).toBe(nodeId);
+  expect(result[0].status).toBe("ok");
+  expect(result[0].flist).toBe(vms.machines[0].flist);
+  expect(result[0].entrypoint).toBe(vms.machines[0].entrypoint);
+  expect(result[0].mounts).toHaveLength(1);
+  expect(result[0].interfaces[0]["network"]).toBe(vms.network.name);
+  expect(result[0].interfaces[0]["ip"]).toContain(splitIP(vms.network.ip_range));
+  expect(result[0].interfaces[0]["ip"]).toMatch(ipRegex);
+  expect(result[0].capacity["cpu"]).toBe(cpu);
+  expect(result[0].capacity["memory"]).toBe(memory * 1024);
+  expect(result[0].planetary).toBeDefined();
+  expect(result[0].publicIP).toBeNull();
+  expect(result[0].description).toBe(description);
+  expect(result[0].mounts[0]["name"]).toBe(diskName);
+  expect(result[0].mounts[0]["size"]).toBe(bytesToGB(diskSize));
+  expect(result[0].mounts[0]["mountPoint"]).toBe(mountPoint);
+  expect(result[0].mounts[0]["state"]).toBe("ok");
+
+  const backends = ["http://[" + result[0].planetary + "]:88"];
+  log(backends);
+
+  //Name Gateway Model
+  const gw: GatewayNameModel = {
+    name: name,
+    node_id: GatewayNode.nodeId,
+    tls_passthrough: tlsPassthrough,
+    backends: backends,
+  };
+
+  const gatewayRes = await gridClient.gateway.deploy_name(gw);
+  log(gatewayRes);
+
+  //Contracts Assertions
+  expect(gatewayRes.contracts.created).toHaveLength(1);
+  expect(gatewayRes.contracts.updated).toHaveLength(0);
+  expect(gatewayRes.contracts.deleted).toHaveLength(0);
+  expect(gatewayRes.contracts.created[0].contractType.nodeContract.nodeId).toBe(GatewayNode.nodeId);
+
+  const gatewayResult = await gridClient.gateway.getObj(gw.name);
+  log(gatewayResult);
+
+  //Gateway Assertions
+  expect(gatewayResult[0].name).toBe(name);
+  expect(gatewayResult[0].status).toBe("ok");
+  expect(gatewayResult[0].type).toContain("name");
+  expect(gatewayResult[0].domain).toContain(name);
+  expect(gatewayResult[0].tls_passthrough).toBe(tlsPassthrough);
+  expect(gatewayResult[0].backends).toStrictEqual(backends);
+
+  const site = "https://" + gatewayResult[0].domain;
+  let reachable = false;
+
+  for (let i = 0; i < 180; i++) {
+    const wait = await setTimeout(5000, "Waiting for gateway to be ready");
+    log(wait);
+
+    await axios
+      .get(site)
+      .then(res => {
+        log("gateway is reachable");
+        log(res.status);
+        log(res.statusText);
+        log(res.data);
+        expect(res.status).toBe(200);
+        expect(res.statusText).toBe("OK");
+        expect(res.data).toContain("Congratulations, you installed Discourse!");
```

Comment:
if you wait for discouse to be totally ready, i think you need to increase test timeout

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/6e463b04-f6da-43ad-bc4f-8f9c45aa1036)

Reply:
and please make sure that test ends by timeout, i noticed some related logs after timeout and running another test but im not sure 
![Screenshot from 2024-06-27 16-25-14](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/62248851/db7b6cdd-0ff1-4fec-8d71-0d67acddfb60)

Reply:
> if you wait for discouse to be totally ready, i think you need to increase test timeout
> 

![image](https://github.com/threefoldtech/tfgrid-sdk-ts/assets/109473873/bfa21a09-79fc-4686-94c5-419549f6a052)

Reply:
Don't forget to add the flag '--runInBand` to run tests in series, or just add the test file name.